docs/guides/distribution/updater.md

```diff
@@ -2,220 +2,128 @@
+- The _public key_, which will be added to your `tauri.conf.json` file later, to validate update artifacts before the installation.
+- The _private key_, which is used to sign your update artifacts and should NEVER be shared with anyone. Also, if you loose this key, you will NOT be able to publish new updates to your current user base. It is crucial to store it in a safe place where you can always access it.
 
-### Built-in dialog
+To generate the keys, you need to use the Tauri CLI:
 
-By default, the updater uses a built-in dialog API from Tauri.
+<Command name="signer generate -w ~/.tauri/myapp.key" />
 
-![New Update](https://i.imgur.com/UMilB5A.png)
+If you are on Windows, you should use `$HOME/.tauri/myapp.key` instead.
 
-The dialog release notes are represented by the update `note` provided by the [server](#server-support).
-If the user accepts, the update is downloaded and installed. Afterward, the user is prompted to restart the application.
+## Tauri configuration
 
-### Javascript API
-
-:::caution
-You need to _disable built-in dialog_ in your [tauri configuration](#configuration); Otherwise, the javascript API will NOT work.
-:::
+Now you need to configure Tauri's updater. To do this, add this to your Tauri config file:
 
-```js
-import { checkUpdate, installUpdate } from '@tauri-apps/api/updater'
-import { relaunch } from '@tauri-apps/api/process'
-try {
-  const { shouldUpdate, manifest } = await checkUpdate()
-  if (shouldUpdate) {
-    // display dialog
-    await installUpdate()
-    // install complete, restart the app
-    await relaunch()
+```json title=tauri.conf.json
+{
+  "tauri": {
+    "updater": {
+      "active": true,
+      "endpoints": [
+        "https://releases.myapp.com/{{target}}/{{arch}}/{{current_version}}"
+      ],
+      "dialog": true,
+      "pubkey": "YOUR_UPDATER_SIGNATURE_PUBKEY_HERE"
+    }
   }
-} catch (error) {
-  console.log(error)
 }
 ```
 
-### Events
-
-:::caution
-
-You need to _disable the built-in dialog_ in your [tauri configuration](#configuration); Otherwise, events aren't emitted.
-
-:::
-
-To know when an update is ready to be installed, you can subscribe to these events:
-
-#### Initialize updater and check if a new version is available
-
-##### If a new version is available, the event `tauri://update-available` is emitted.
-
-Event: `tauri://update`
-
-#### Rust
-
-```rust
-window.emit("tauri://update".to_string(), None);
-```
-
-#### Javascript
-
-```js
-import { emit } from '@tauri-apps/api/event'
-emit('tauri://update')
-```
-
-#### Listen to the New Update Available event
-
-Event: `tauri://update-available`
-
-Emitted data:
-
-```
-version    Version announced by the server
-date       Date announced by the server
-body       Note announced by the server
-```
-
-#### Rust
+The required keys are `"active"`, `"endpoints"` and `"pubkey"` to enable the updater. `"dialog"` is optional and will default to `true` if not set.
```

Comment:
There's a LOT of detail for what each of these keys are, the values, etc. I wonder if this information is better to have in the reference docs and link to it there? Helps with future maintainability as well

Reply:
Yes: https://tauri.app/v1/api/config#updaterconfig.
I don't think removing this section in favor of a link would be beneficial here though to be honest. I added a link to the config object above so hopefully that's enough of a cross ref?

Reply:
Sounds good to me